src/window.mm

```diff
@@ -148,6 +148,40 @@ - (void)notified:(const std::string &)note withData:(const msgpack::object &)upd
         AppDelegate *app = (AppDelegate *)[NSApp delegate];
         [app newWindow];
     }
+    else if (note == "neovim.app.openFile") {
+        NSArray *files = [mMainView showFileOpenDialog];
+        if (files != nil) {
+            std::vector<char *> args;
+
+            /* open in tabs if more than one file was selected */
+            if ([files count] > 1)
+                args.push_back(const_cast<char *>("-p"));
+
+            for (NSURL *url in files) 
+                args.push_back(const_cast<char *>([[url path] UTF8String]));
+
+            AppDelegate *app = (AppDelegate *)[NSApp delegate];
+            [app newWindowWithArgs:args];
+        }
+    }
+    else if (note == "neovim.app.saveFile") {
+        NSURL *file = [mMainView showFileSaveDialog];
+        if (file != nil)
+        {
+            std::stringstream cmd;
+            cmd << "w " << [[file path] UTF8String];
```

Comment:
Yes. Opens and saves files with weird characters in them. even tried files with japanese characters as names.